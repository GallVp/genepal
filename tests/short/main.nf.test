nextflow_pipeline {

    name "Test for short assemblies"
    script "main.nf"

    test("short fasta - error - terminate gracefully") {

        when {
            params {
                input               = "$baseDir/tests/short/assemblysheet.csv"
                protein_evidence    = "https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/sarscov2/genome/proteome.fasta.gz"
                busco_skip          = true
                outdir              = "$outputDir"
            }
        }

        then {
            def stable_path = getAllFilesFromDir(params.outdir, false, ['pipeline_info/*.{html,json,txt,yml}'], null, ['**'])

            assertAll(
                { assert ! workflow.success },
                { assert workflow.stdout.toString().contains("The assembly represented by tag 'sarscov2' is only 8823 bytes. The minimum allowed size is 1 MB!") },
                { assert snapshot(
                    [
                        'successful tasks': workflow.trace.succeeded().size(),
                        'versions': removeNextflowVersion("$outputDir/pipeline_info/genepal_software_mqc_versions.yml"),
                        'stable paths': stable_path
                    ]
                ).match() }
            )
        }

    }

}
