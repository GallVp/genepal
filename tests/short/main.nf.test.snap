{
    "short fasta - error - terminate gracefully": {
        "content": [
            {
                "successful tasks": 0,
                "versions": {
                    "Workflow": {
                        "plant-food-research-open/genepal": "v0.6.1dev"
                    }
                },
                "stable paths": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-11T12:26:15.565583"
    }
}